### DISCUSSION TITLE
Create a Godwoken Mainnet CKB Faucet to solve the `gas fee paradox` for new EOA accounts

### DISCUSSION THREAD

**Discussion by Flouse** [nervosnetwork/godwoken#1072] [Ideas] [answered]
### (1) wuxian from .bit community:
When people sign up for their.bit, ETH/BNB/TRX/Polygon and DogeCoin can also be paid, but they all need to pay the GAS fee. While using CKB to buy .bit is more cost-effective. Hundreds of users want to register .bit ID by paying CKB but **don't know where to buy CKB. So far, users have had to go to CEX, like Binance, or third parties that work with .bit to buy CKB.

I am wondering if we can swap CKB in DEX though Godwoken.

### (2) @alejandroRbit:
If someone bridges some USDC from Celer cBridge to Godwoken, he can't do swap transactions because CKB is required to pay gas fee.

But he doesn't have any CKB at the beginning.

### (A) Godwoken Mainnet CKB Faucet
To help address this pain point, we can create a Godwoken Mainnet CKB Faucet solving the `gas fee paradox` for new EOA accounts. It is possible to airdrop a small amount of CKB to new users' addresses ((EOA) when they bridge assets to Godwoken, to support initial transaction fees.

#### **Rules**: What kind of account can claim this CKB faucet?
1. The account must contain assets such as USDC/USDT bridged from Celer cBridge or Multichain.
2. Each account can only receive the faucet **_once_** with a limit of 30 CKB, which is only enough for about 2 [YokaiSwap](https://www.yokaiswap.com/) or [Hadouken](https://app.hadouken.finance/swap/) DeFi transactions.
3. (Optional) A rule similar to https://goerlifaucet.com
   > => “To receive Goerli ETH, your wallet must have a minimum balance of 0.001 ETH on Ethereum mainnet."

## Discussion
Do you think it is achievable way? Or are there other solutions in your mind?

**Comment by alejandroRbit** [comment]
I'm glad to see this discussion.

I think that first we would have to differentiate the various problems that we are faced with here, which are among others:

1º User experience, functionality and usability of the bridges.
2º Acquire CKB and other assets in Godwoken without going through a CEX, that is, in a decentralized way.
3º Acquire L1 assets directly from L2.

### **User experience, functionality and usability of bridges**

Currently the only really usable bridge for usability and liquidity is ForceBridge, which curiously is the least user friendly since it is an L1 - L2 bridge and therefore makes use of the ORU challenge time, which is extremely damaging to the user experience and Constant source of complaints, and that's without taking into account other problems like ForceBridge fees when going back to Ethereum for example. The conclusion to be drawn is that users generally prefer to use L2 - L2 bridges, which is precisely where they should live. But both Celer and Multichain are illiquid and unusable due to gas fee paradox.

Something that would improve the user experience is CKB wrapped in other networks like Ethereum (and its L2), since they would swap in their home network and transfer CKB through the bridge, in fact I think I remember that one of ForceBridge's priorities was have wCKB on Ethereum/BSC.

### **Acquire CKB and other assets in Godwoken without going through a CEX, that is, in a decentralized way**

Currently the most direct way to acquire CKB in a decentralized way is as follows:

- Use ForceBridge to bring for example USDC/ETH
- Deposit in L2 (these two steps can be done through application interfaces such as Hadouken or YokaiSwap)
- Swap USDC/ETH for CKB in L2. Thanks to the fact that FB converts part of the funds into CKB, it is possible to swap, that is, FB does not suffer from gas fee paradox (reference: https://discord.com/channels/930587710673006623/930587711885152328/1092578917400973373)

This is an important point as it is one of the strongest use cases right now for Godwoken. It is the only way to buy CKB without going through third parties.

### **Acquire L1 assets directly from L2**

Originally what Wuxian was asking for is to be able to buy a .bit account from Godwoken (reference: https://discord.com/channels/1006631225659773059/1006991885203669153/1087398947372400821). And in my opinion it makes a lot of sense since transactional users live in L2, but use L1 products (we'll use .bit as an example but the goal is to be able to buy any L1 asset from L2). In this way, the user experience is improved by eliminating extrasteps.

***

Taking all this into account, we can answer the following questions about the proposed solution:

**Improve the user experience and usability of the bridges?**
Yes, it makes usable Celer and Multichain bridges, L2-L2 bridges which are the most user friendly bridges, and those that transactional users should use.

**Allows/improves acquiring CKB in a decentralized way?**
Yes, it opens up two new ways to acquire CKB without going through a CEX.

**Allows users to purchase L1 assets directly from L2?**
No, the problem still persists, I have to make a withdrawal to L1, and wait approximately 9 days, or I have to buy CKB on an exchange and send to did balance to buy a .bit account or other L1 asset.

***

**About the proposed solution** I think is good and desirable since it allows to use Celer and Multichain effectively, improves the user experience and allows users to buy CKB in a decentralized way (without going through a CEX) through more options and therefore I think it should be implemented. But I also think that it can be improved, that's why I have asked CKBased.bit to intervene in the discussion (and according to what he has told me, he is going to do it during the week). He knows a lot about it from his experience at Arbitrum and has several ideas to improve the experience.

Regarding buying L1 assets from L2, I think the solution improves the user experience, accessibility and usability of Godwoken, which is great but it doesn't solve the main problem, which is that if you can't buy them directly from Godwoken, the user it has to go through the challenge time, or rely on third parties to acquire CKB, which is not ideal. The solution to this problem is perhaps much more complex and technical (maybe through light-godwoken?) than the one associated with the gas fee paradox, but I recommend keeping it in mind for future implementations and features.

**Comment by Flouse** [comment, marked as answer]
# Godwoken Faucet
GitHub Repository: https://github.com/godwokenrises/godwoken-faucet

## Deployments
- testnet
   https://faucet-testnet.godwoken.io
- mainnet
   https://faucet-mainnet.godwoken.io
